src/rpc/core_rpc_server.cpp

```diff
@@ -2966,5 +2966,89 @@ namespace cryptonote
     res.checkpoints = db.get_checkpoints_range(req.start_height, req.end_height);
     return true;
   }
+  //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_get_service_nodes_state_changes(const COMMAND_RPC_GET_SN_STATE_CHANGES::request& req, COMMAND_RPC_GET_SN_STATE_CHANGES::response& res, epee::json_rpc::error& error_resp, const connection_context *ctx)
+  {
+    using blob_t = cryptonote::blobdata;
+    using block_pair_t = std::pair<blob_t, block>;
+    std::vector<block_pair_t> blocks;
+    std::vector<blob_t> txs;
+
+    const auto& db = m_core.get_blockchain_storage();
+    const uint64_t current_height = db.get_current_blockchain_height();
+
+    uint64_t end_height;
+    if (req.end_height == COMMAND_RPC_GET_SN_STATE_CHANGES::HEIGHT_SENTINEL_VALUE) {
+      end_height = current_height;
+    } else {
+      end_height = req.end_height;
+    }
+
+    if (!db.get_blocks(req.start_height, end_height - req.start_height, blocks, txs)) {
```

Comment:
Should handle the case where req.start_height > end_height

Reply:
Done.
Also added a `+ 1` to ensure `count` is set to `1` when `start == end`, i.e. when we want to get events for a specific block...